src/views/documents/utils/QuerySortDropdown.vue

```diff
@@ -0,0 +1,148 @@
+          :class="{ 'has-text-weight-bold': field == sortField }"
+          @click="toggle(field)"
+        >
+          {{ $gettext(field) }}
+        </span>
+        <span v-if="field == sortField" class="has-cursor-pointer">
+          <fa-icon v-if="reversed" icon="sort-amount-up" @click="setSort(field)" />
+          <fa-icon v-if="!reversed" icon="sort-amount-down-alt" @click="setSort('-' + field)" />
+        </span>
+      </div>
+    </div>
+    <div v-else class="content">
+      <div class="dropdown-item is-italic">
+        <span v-if="documentType === 'waypoint'" v-translate> Please set a waypoint type to enable sort feature </span>
+        <span v-translate> Please set an activity to enable sort feature </span>
+      </div>
+    </div>
+  </dropdown-button>
+</template>
+
+<script>
+import urlMixin from './url-mixin';
+
+import constants from '@/js/constants';
+
+export default {
+  mixins: [urlMixin],
+
+  data() {
+    return {
+      sortField: null,
+      reversed: null,
+      complexSort: false,
+      sortableFields: [],
+    };
+  },
+
+  watch: {
+    $route: {
+      handler: 'updateFromRoute',
+      immediate: true,
+    },
+  },
+
+  methods: {
+    setAsComplexSort() {
+      this.sortField = null;
+      this.reversed = null;
+      this.complexSort = true;
+    },
+
+    updateFromRoute() {
+      const sortableFields = [];
+
+      for (const field of Object.values(constants.objectDefinitions[this.documentType].fields)) {
+        if (field.sortable && this.fieldIsVisible(field)) {
+          sortableFields.push(field.name);
+        }
+      }
+
+      this.sortableFields = sortableFields;
+
+      if (!this.$route.query.sort) {
+        this.sortField = null;
+        this.reversed = null;
+        this.complexSort = false;
+        return;
+      }
+
+      if (this.$route.query.sort.includes(',')) {
+        this.setAsComplexSort();
+        return;
+      }
+
+      let sortField = this.$route.query.sort;
+      if (sortField.startsWith('-')) {
+        sortField = sortField.substring(1);
+        this.reversed = true;
+      } else {
+        this.reversed = false;
+      }
+
+      if (!this.sortableFields.includes(sortField)) {
+        this.setAsComplexSort();
+        return;
+      }
+
+      this.complexSort = false;
+      this.sortField = sortField;
+    },
+
+    toggle(field) {
+      this.setSort(this.sortField === field ? undefined : field);
+    },
+
+    clear() {
+      this.setSort(undefined);
+    },
+
+    setSort(field) {
+      const query = Object.assign({}, this.$route.query);
+      query.sort = field;
+      this.$router.push({ query });
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.content {
+  min-width: 250px;
+}
+
+.sort-item {
+  display: flex !important;
+}
+.sort-label {
+  flex-grow: 1;
+  padding-right: 2em;
+}
+
+@media screen and (max-width: $tablet) {
+  .button.is-small-mobile {
+    border-radius: 2px;
+    font-size: 0.7857rem;
```

Comment:
another hard coded magic value... 🙁

Reply:
I copy pasted from `query-items`, but I'm the one to blame :D 
https://github.com/c2corg/c2c_ui/commit/3814fd1a4ceb45def6b207c2b254ffb7069f3d2e#diff-536a276a9dca18e2ac7816564a4da972e3efed4a191c1547df3148e2bbbe07d9R282

Fixed, I've used bulma helper.